libgc/os_dep.c

```diff
@@ -32,7 +32,7 @@
       /* prototypes, so we have to include the top-level sigcontext.h to    */
       /* make sure the former gets defined to be the latter if appropriate. */
 #     include <features.h>
-#     if 2 <= __GLIBC__
+#     if 2 <= __GLIBC__ || !defined(__GLIBC__)
```

Comment:
This seems to contradict the other change you made in exceptions-arm.c. There you include asm/sigcontext.h if `__GLIBC__` is defined, but here it's the other way round (it'll fall through to the else and include it when `__GLIBC__` is **not** defined).